test.py

```diff
@@ -14,24 +14,37 @@
 
 import chardet
 
+EQUIVALENT_ENCODINGS = {'latin1': 'windows-1252', 'windows-1254': 'iso-8859-9'}
+LD_EQUIVALENT_ENCODINGS = {'hungarian': {'iso-8859-2': 'windows-1250'}, 'romanian': {'iso-8859-2': 'windows-1250'}} # Language (alphabet) dependend equivalency
+MISSING_ENCODINGS = set(['iso-8859-6', 'windows-1256'])
 
-EQUIVALENT_ENCODINGS = {'latin1': 'windows-1252'}
-# TODO: Restore Hungarian encodings (iso-8859-2 and windows-1250) after we
-#       retrain model.
-MISSING_ENCODINGS = set(['iso-8859-2', 'iso-8859-6', 'windows-1250',
-                         'windows-1254', 'windows-1256'])
+import re
 
+remove_whole = ['language', 'generator', 'copyright', 'link', 'url', 'pubDate', 'lastBuildDate', 'category', 'id', 'guid', 'comments', 
+		'published', 'updated', 'issued', 'modified', 'created', 'datePosted', 'dc:[a-z]*'] 
 
-def check_file_encoding(file_name, encoding):
-    """ Ensure that we detect the encoding for file_name correctly. """
+def remove_tags(buf):
```

Comment:
OK, but there is really very BIG problem:
Which language is correct for detection in a text with many xml tags (= many english words).
There are many xml files where the count of the english words (because of tags) is greather than count of "normal" words. In such case (by propability) the Latin1 is always the winner, and it isn't correct. 

Text in the hungarian language can't contain many english words inside detected text. For example xml files can have more english words because of tag names.
The second problem arises if the hungarian text has many uppercase sentences.